bug: test_lambda_functions_created expects 2 Lambda functions but API stack now has 3

## Bug

`tests/test_api_stack.py::test_lambda_functions_created` fails with:

```
AssertionError: Expected 2 resources of type AWS::Lambda::Function but found 3
```

### Root Cause

A third Lambda function was added to the API stack (likely a new endpoint or authorizer), but the CDK test at `tests/test_api_stack.py:29` still asserts:

```python
template.resource_count_is("AWS::Lambda::Function", 2)
```

### Fix

Update the assertion to match the actual number of Lambda functions in the stack, or make the test count dynamically.

### Impact

- 1 test failing in every CI run (30/31 pass)
- Does not block deploys currently but masks real regressions

### Seen In

- PR #58 CI run: https://github.com/barakcaf/runmaprepeat/actions/runs/23514654214
- Pre-existing before that PR

--- a/infra/tests/test_api_stack.py
+++ b/infra/tests/test_api_stack.py
@@ -26,7 +26,7 @@ def test_rest_api_created() -> None:
 
 def test_lambda_functions_created() -> None:
     template = _create_api_stack()
-    template.resource_count_is("AWS::Lambda::Function", 2)
+    template.resource_count_is("AWS::Lambda::Function", 3)
 
 
 def test_lambda_uses_arm64() -> None:

